src/main/java/io/lakefs/iceberg/LakeFSReporter.java

```diff
@@ -0,0 +1,81 @@
+package io.lakefs.iceberg;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.hadoop.conf.Configuration;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LakeFSReporter {
+    private URI lakeFSEndpoint;
+    private Configuration hadoopConfig;
+    private HttpClient httpClient;
+    private final String reportClient = "iceberg-catalog/" + getClass().getPackage().getImplementationVersion();
+
+    private LakeFSReporter(){}
+    public LakeFSReporter(Configuration hadoopConfig) {
+        String lakeFSServerURL = hadoopConfig.get("fs.s3a.endpoint");
+        if (lakeFSServerURL.endsWith("/")) {
+            lakeFSServerURL = lakeFSServerURL.substring(lakeFSServerURL.length()-1);
+        }
+        lakeFSServerURL += "/api/v1/statistics";
+        this.lakeFSEndpoint = URI.create(lakeFSServerURL);
+        this.hadoopConfig = hadoopConfig;
+        this.httpClient = HttpClient.newHttpClient();
+    }
+
+    public void logOp(String op) {
+        Map<String, Object> reportMap = new HashMap<>() {
+            {
+                put("class", "integration");
+                put("name", op);
+                put("count", 1);
+            }
+        };
+        try {
+            log(reportMap);
```

Comment:
(probably messed up the indentation)

```suggestion
        try {
            log(Map.of("class", "integration",
                                "name", op,
                                "count", 1));
```

Reply:
done